src/components/DnsSettingsForm.js

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) [2020] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useState } from 'react';
+import cockpit from 'cockpit';
+import {
+    FormGroup,
+    TextInput
+} from '@patternfly/react-core';
+import { useNetworkDispatch, updateDnsSettings } from '../context/network';
+import ModalForm from './ModalForm';
+import IPInput from './IPInput';
+
+const _ = cockpit.gettext;
+
+const DnsSettingsForm = ({ isOpen, onClose, dns }) => {
+    const dispatch = useNetworkDispatch();
+    const originalNameServers = (dns?.nameServers || "").split(" ");
```

Comment:
Destructuring can _help_ a bit here

```js
const nameServers = (dns?.namesrvers ||  "");
const [ firstNameServer = "", secondNameServer = "", thirdNameServer = "" ] = nameServers.split(" ")
```

Although we still having the `dns?nameServers || ""` statement there. Which could be removed in an _ugly_ way, 

```js
const DnsSettingsForm = ({ isOpen, onClose, { policy = "", nameServers = "",  searchList = "" } = {}}) => {
    const dispatch = useNetworkDispatch();
    const [ firstNameServer = "", secondNameServer = "", thirdNameServer = "" ] = nameServers.split(" ")
    const [nameserver1, setNameserver1] = useState(firstNameServer);
    const [nameserver2, setNameserver2] = useState(secondNameServer);
    const [nameserver3, setNameserver3] = useState(thirdNameServer);
    const [policy, setPolicy] = useState(policy);
    const [searchList, setSearchList] = useState(searchList);

   ...
```

Too much destructuring, I know. It would be better that the `dns` prop were already _sanitized_ at this point. But it could be addressed in other iteration.